research/upgrade_interviews/index.md

```diff
@@ -0,0 +1,56 @@
+- What was the reason you last upgraded your cluster?
+- Is there any internal process that needs to happen before upgrading clusters?
+- How do you track the progress of an upgrade?
+
+## Findings
+
+### The upgrade ‘job map’
+Our interviews uncovered fundamental aspects of the upgrade process for a computing platform. There are some OpenShift 4 and 3 details, but these are good aspects to keep in mind when designing around the upgrade process for any platform. We borrowed the idea of a job map from the jobs-to-be-done framework to explain the process.
+
+![](img/upgrade_job_map.png)
+
+We found a few particular job map aspects worth writing out.
+
+In steps 3-8, all components are repeated in some form because all upgrades are multi-cluster: they move from development towards production environments. This means upgrades take place at least twice, if not more than that, to fully upgrade a company’s platform infrastructure.
+
+In step 3 (prepare), almost all customers we talked to have to approve upgrades via a change management board (for version 3 and 4). This manual process is something not likely to be waved away because of new technological abilities because it’s often formally written into contracts or company policy.
+
+In step 4 (conclude), alerts are turned off manually. Alerts tend to fire off at a very high rate and overload existing alert channels. The alerting state during the upgrade was often seen as overwhelming and not useful.
+
+In step 5 (execute), this is the step, the upgrade! It’s interesting that we almost spent little time on this moment. In v4, it’s a button click and in v3 it’s beginning an Ansible playbook. This was fairly simple and a reminder that the context around a singular feature is often critical.
+
+After beginning the upgrade, step 6 (monitor) begins. When checking out alerts manually, it was a challenging process to assess which alerts needed action and which would heal on their own. If something did break, the operator would often hear of this from a developer or another platform reliant group, like the logging team.
+
+### General findings
+
+One of the biggest desired emotions to avoid is the feeling of ‘flakiness’ or like upgrading is a ‘Hail Mary’. This is complicated by the fact that success in an upgrade isn’t purely defined by what an operator controls in the platform. Even if the cluster itself upgrades without a problem, any application downtime is a problem. A key gap is how Administrators could check beforehand if developer applications are not sufficiently resilient and may go down in the upgrade. This process is manual right now and often only discovered after something goes wrong. The platform exists as a means to an end - the critical function is to support developer applications.  To quote a user, "my experience depends on what applications are running on OpenShift. And their application might not be able to handle the restart or handle a restart as gracefully as they would like."
```

Comment:
Could we reword the first sentence? 

What about something like "During the upgrade process, the Operator wants to feel in control. The user experience should avoid any approach that leads them to the feeling of ‘flakiness’ or like upgrading is a ‘Hail Mary’...."

Reply:
Administrator is a more classic term, but I think Operator aligns better with the company terminology. I personally struggle with the term Operator and having 'operators' be a main function of OpenShift, but that may be too far gone to adjust. I'll switch over to Operator.